### ISSUE TITLE
Error vs sentinel

### ISSUE THREAD

**Issue by jtrakk** [gkappler/CombinedParsers.jl#23]
Sometimes people say that exceptions should only be raised for unexpected states in the program, but that a nonmatching parse is often an expected state so it should just return a sentinel value instead of raising an error. What do you think?

**Comment by gkappler** [comment]
I agree. The parse method throws an error to match behaviour of `Base.parse(Int,"abc")`. Please refer to `tryparse` instead, that returns nothing if there is no match.

Yet your question makes me wonder whether instead returning `NoMatch()` singleton would be clearer as `nothing` might be a intended parsing result. What do you think?

**Comment by jtrakk** [comment]
I agree returning `NoMatch()` would be less ambiguous in cases where `nothing` is a valid parse result. I suppose theoretically `NoMatch()` could be an intended parsing result, in which case `tryparse` could take a `sentinel` argument.
